pkg/backend/oss.go

```diff
@@ -63,114 +61,104 @@ func newOSSBackend(rawConfig []byte) (*OSSBackend, error) {
 	return &OSSBackend{
 		objectPrefix: objectPrefix,
 		bucket:       bucket,
+		forcePush:    forcePush,
 	}, nil
 }
 
-// Ported from https://github.com/aliyun/aliyun-oss-go-sdk/blob/c82fb81e272d84f716d3f13c36fe0542a49adfeb/oss/utils.go#L207.
-func splitBlobByPartNum(blobSize int64, chunkNum int) ([]oss.FileChunk, error) {
-	if chunkNum <= 0 || chunkNum > 10000 {
-		return nil, errors.New("chunkNum invalid")
+func splitFileByPartSize(blobSize, chunkSize int64) ([]oss.FileChunk, error) {
+	if chunkSize <= 0 {
+		return nil, errors.New("invalid chunk size")
 	}
 
-	if int64(chunkNum) > blobSize {
-		return nil, errors.New("oss: chunkNum invalid")
+	var chunkN = blobSize / chunkSize
+	if chunkN >= 10000 {
+		return nil, errors.New("too many parts, please increase chunk size")
 	}
 
 	var chunks []oss.FileChunk
 	var chunk = oss.FileChunk{}
-	var chunkN = (int64)(chunkNum)
 	for i := int64(0); i < chunkN; i++ {
 		chunk.Number = int(i + 1)
```

Comment:
Why is there a offset `1` between chunk number and the index?

Reply:
It's defined by `oss.FileChunk`.